### DISCUSSION TITLE
Improve documentation for using renovate GitHub Plugin with private maven repositories

### DISCUSSION THREAD

**Discussion by ahoeing** [renovatebot/renovate#9202] [Request Help] [answered]
**What would you like Renovate to be able to do?**

I read that renovate is able to scan private repositories. But I have not found any documentation about how to configure the credentials in GitHub.  All I found was: https://docs.renovatebot.com/java/

Maybe you can Improve the documentation or maybe it is not possible for the GitHub plugin?

**Comment by HonkingGoose** [comment]
> I read that renovate is able to scan private repositories.

Renovate can scan private repositories just fine if you're using the GitHub hosted app, see: https://docs.renovatebot.com/install-github-app/

How are you using Renovate bot right now?

---

I'm moving this to the discussion tab, as it seems more of configuration help problem than a problem with the current docs.

**Reply by ahoeing** [reply]
Thanks for the fast response. I expressed myself somehow missleading.

I want to scan a private maven repository (private nexus) to get information about updated versions from there.

**Reply by HonkingGoose** [reply]
> I want to scan a private maven repository (private nexus) to get information about updated versions from there.

Oh that's a different problem than what I could understand from your description, I thought you had issues with telling Renovate bot to run on a private repository.

I'll let the Renovate developers help you with getting your config up and running. 😉 
If it turns out we need to improve the docs, I'll make a issue to track that work.

**Comment by rarkins** [comment, marked as answer]
You still need `hostRules` but you can add them to `renovate.json` instead in JSON format, using `encrypted`. Here's a similar older question I found: https://github.com/renovatebot/config-help/issues/426

Any PR to improve the documentation on this would be very welcome

**Reply by HonkingGoose** [reply]
I've opened issue #9205 to track the docs changes we need.

@ahoeing I'm really sorry for moving this to a discussion too soon, as your issue was actually in the right place... 🙈

**Reply by rarkins** [reply]
I would prefer people raise discussions on this topic anyway, and we decide if it's worth creating an issue for every missing or incomplete documentation or just fix it directly (I do not believe all docs PRs need an issue). Most open source projects have hundreds of areas they could improve documentation and the best community contribution would be for people to contribute a PR and not weigh down the project with issues for someone else to do

**Reply by HonkingGoose** [reply]
> I would prefer people raise discussions on this topic anyway, and we decide if it's worth creating an issue for every missing or incomplete documentation or just fix it directly

Do you want a PR from me to redirect docs questions from users to the discussions tab? That way you could do a first "triage" of whether or not you want a issue for the problem or not?

> (I do not believe all docs PRs need an issue). Most open source projects have hundreds of areas they could improve documentation and the best community contribution would be for people to contribute a PR and not weigh down the project with issues for someone else to do

I'm confused, does this mean you want to get rid of all docs type issues, or just the docs type issues that somebody else needs to fix?
Do you want me to stop making `type:docs` issue on the repository?
Can you maybe simplify this for me? I'm really confused on what you want to be different/changed here.

**Reply by rarkins** [reply]
For anyone who contributes docs improvements, it's fine to just PR directly and keep noise lower than if there's an issue too. However it's also fine to create issues as a kind of "personal reminder" to do later, or if it needs discussion. In other words, whatever is most efficient/convenient for you as the contributor.
